MTL fix the memory information parsing

miczyg1 opened the pull request:
Fix for the issue with 0MB RAM described here: https://github.com/Dasharo/dasharo-issues/issues/1259

miczyg1 committed:
soc/intel/meteorlake/romstage: Skip no present memory controllers

Don't parse memory controllers that are not present or disabled.
Can save a little boot time. Verified by aprsing the raw memory hob
that FSP corrrectly marks the controllers as not present with all
underlying channels and dimms.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 committed:
vc/intel/fsp/fsp2_0/iot/meteorlake/MemInfoHob.h: Match content with used FSP

Fixes the issue where the were no DIMMs found in FSP HOB and as a result
SMBIOS memory information was not populated.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

mkopec reviewed (dismissed):
The changes to the header look to be the same as in https://github.com/Dasharo/fwbuild/blob/main/v540tu/uefi/v0.9.0/bios/FWBUILD , so LGTM

mkopec commented:
alternatively, we could cherrypick https://review.coreboot.org/c/coreboot/+/84656

miczyg1 commented:
I forgot to update configs to get rid of FSP paths from 3rdparty/blobs

miczyg1 committed:
configs/config.novacustom_mtl: Update configs

The configs had the debugging enabled by default which made the
boot time unnecessarily longer.

Also update the FSP paths to use publicly available components.

Signed-off-by: Michał Żygowski <michal.zygowski@3mdeb.com>

miczyg1 commented:
@mkopec please check https://github.com/Dasharo/coreboot/pull/628/commits/e23c6d20711fcb5d1dcf4c9ce83443099e655fd7. If the build passes then we can merge I think.
